sc2ts/utils.py

```diff
@@ -558,6 +559,107 @@ def get_imputed_pango(u):
 
         return pd.DataFrame(recombinants)
 
+    def export_recombination_node_breakpoints(self):
+        """
+        Make a dataframe with one row per recombination node breakpoint
+        Return info about the path to the mrca of left and right parents when travelling
+        up the tree to the left of the the breakpoint and to the right of the breakpoint.
+        
+        You can only list the breakpoints in recombination nodes that have 2 parents by
+        doing e.g. df.drop_duplicates('node', keep=False)
+        
+        """
+        # Note that it is more efficient to find the MRCA nodes in a single batch
+        # for all breakpoints, reusing the trees, rather than getting a new tree
+        # for each recombination node
+
+        # Sort recombinants by breakpoint location
+        forward_breakpoints = collections.defaultdict(list)
+        meta = self.nodes_metadata
+        for u in self.recombinants:
+            md = meta[u]
+            row = md["match_info"]
+            assert len(row) == 2
+            assert row[0]["strain"] == row[1]["strain"]
+            matches = {}
+            for record in row:
+                matches[record["direction"]] = Match(
+                    breakpoints=record["breakpoints"],
+                    parents=record["parents"],
+                    mutations=record["mutations"],
+                )
+            rec = Recombinant(row[0]["strain"], matches, node=u)
+            breaks = rec.matches["forward"].breakpoints[1:-1]
+            for i in range(len(breaks)):
+                forward_breakpoints[breaks[i]].append(
+                    (u, rec.is_hmm_consistent(), rec.matches["forward"].parents[i:i+2])
+                )
+        sorted_breakpoints = sorted(forward_breakpoints.keys())
+        tree_a_indexes = np.searchsorted(
+            self.ts.breakpoints(as_array=True),
+            sorted_breakpoints,
+            side='left',
+        ) - 1
+        tree_a_indexes = np.searchsorted(
+            self.ts.breakpoints(as_array=True),
+            sorted_breakpoints,
+            side='right',
+        ) - 1
+
+        causal_sample_map = get_recombinant_samples(self.ts)
```

Comment:
This function is left over from an earlier case when we didn't have the extra match info metadata. You can just take the ``strain`` from the Recombinant class above (which is a causal sample) and make things a bit simpler.

Reply:
Hmm, I don't quite understand here (but I didn't before either, I think!). Do you mean that *every* recombination node is the "cause" of a particular (named) strain? That seems a bit weird to me. For instance, if you have a re node that creates e.g. XA, and then one of its children is also a recombination node, does that child also get labelled a "causal XA" node? It seems to me that it shouldn't do (because it is nested within the original XA generation process), but perhaps it does under any strategy.

Reply:
It's the other way around, sample (strain, in SARS-Cov-2 parlance) ``a`` causes recombination node ``b`` to be created because it had a copying path consisting of more than one edge. Here ``strain`` is the sample label, not the Pango *lineage* status

Reply:
We're not saying anything is causal of the the pango X lineages - we're just saying that a particular sample is causal for a particular recombination node. Reasoning about the X lineages is separate.

Reply:
Ah, right. I need to be careful in my terminology here. Your phrasing "a particular sample is causal for a particular recombination node" is a great one.

Reply:
I didn't make this improvement in the PR, but I think I need the Pango lineage of the strain, and the `strain` value, which is text, would have to be mapped back to a node ID to get that out anyway, so it's not as easy as that. As long as it's not wrong at the moment, I'll keep it as it but put a comment in the code.